### PR TITLE
feat: wormhole bridging for governance

### PR DESCRIPTION
**Pull request by kupermind**
- Wormhole bridging for governance.

### CODE REVIEW

File: contracts/bridges/WormholeMessenger.sol [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    // value (12 bytes or 96 bits) and the payload size (4 bytes or 32 bits)
+    uint256 public constant DEFAULT_DATA_LENGTH = 36;
+    // L2 Wormhole Relayer address that receives the message across the bridge from the source L1 network
+    address public immutable wormholeRelayer;
+    // Source governor chain Id
+    uint16 public immutable sourceGovernorChainId;
+    // Source governor address on L1 that is authorized to propagate the transaction execution across the bridge
+    address public sourceGovernor;
+    // Mapping of delivery hashes
+    mapping(bytes32 => bool) public mapDeliveryHashes;
```
Comment (kupermind) [inline contracts/bridges/WormholeMessenger.sol:69]: These are two major differences for the wormhole contract - we need to check the source chain Id, and the delivery hash such that it does not repeat.

File: package.json [hunk 2]
```diff
@@ -21,34 +21,34 @@
...
-    "@nomiclabs/hardhat-ethers": "^2.2.2",
-    "@nomiclabs/hardhat-etherscan": "^3.1.6",
-    "@openzeppelin/contracts": "=4.8.3",
-    "@typechain/ethers-v5": "^10.2.0",
-    "@typechain/hardhat": "^6.1.5",
-    "@types/mocha": "^10.0.1",
-    "chai": "^4.3.7",
-    "eslint": "^8.34.0",
+    "@nomicfoundation/hardhat-network-helpers": "^1.0.9",
+    "@nomicfoundation/hardhat-toolbox": "^2.0.2",
```
Comment (kupermind) [inline package.json:25]: Updating hardhat packages such that the default EVM is Paris, and we don't have issues with the deployment on networks that don't deal with `PUSH0` yet.

**Review by 77ph — DISMISSED** [review]

File: scripts/deployment/bridges/wormhole/test/globals_celo.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"contractVerification":true,"useLedger":false,"derivationPath":"m/44'/60'/2'/0/0","providerName":"celoAlfajores","gasPriceInGwei":"10","networkURL":"https://alfajores-forno.celo-testnet.org","sourceGovernorChainId":"10002","L1WormholeRelayerAddress":"0x7B1bD7a6b4E61c2a123AC6BC2cbfC614437D0470","timelockAddress":"0x14CF2e543AB75B321bcf84C3AcC88d570Ccf9106","L2WormholeRelayerAddress":"0x306B68267Deb7c5DfCDa3619E22E9Ca39C374f84","wormholeMessengerAddress":"0x945550dECe7E40ae70C6ebf5699637927eAF13E9","childMockERC20Address":"0x34235f9D447f9F54167e2Ac7A0F4283cB3fAD669"}
```
[the remark this answers is not in the document]
Comment (mariapiamo) [reply scripts/deployment/bridges/wormhole/test/globals_celo.json:1]: ok, thanks. But then in scripts/deployment/bridges/wormhole/globals_celo_mainnet.json we should have sourceGovernorChainId":"1" ??

[the remark this answers is not in the document]
Comment (77ph) [reply scripts/deployment/bridges/wormhole/test/globals_celo.json:1]: 2

**Review by mariapiamo — APPROVED** [review]

**Review by mariapiamo — APPROVED** [review]

**Review by 77ph — APPROVED** [review]

**Pull request merged**